cmd/tailscale/cli/serve_test.go

```diff
@@ -70,17 +71,17 @@ func TestServeConfigMutations(t *testing.T) {
 		command: cmd("funnel on"),
 		want:    &ipn.ServeConfig{AllowFunnel: map[ipn.HostPort]bool{"foo.test.ts.net:443": true}},
 	})
-	add(step{
+	add(step{ // want no change
 		command: cmd("funnel on"),
-		want:    nil, // nothing to save
+		want:    &ipn.ServeConfig{AllowFunnel: map[ipn.HostPort]bool{"foo.test.ts.net:443": true}},
```

Comment:
This is the behavior change I was warning about. Line 58 above says "non-nil means we want a save of this value". You even added "// want no change" on line 74 but you had to fill this in to make this test pass, when it should be nil to mean no save occurred.

You could make your `fakeLocalServeClient` Set method also increment a counter and then the test can check whether the counter was incremented. If not, then nil out the "got" value before comparing to want.

Reply:
That works. I should've asked before changing it in the last push. I misunderstood the feedback.

Reply:
Implemented count as suggested below.